File: app/src/main/java/com/example/focusstartsecondpart/features/events/data/network/EventsLoaderImpl.java
package com.example.focusstartsecondpart.features.events.data.network;

import com.example.focusstartsecondpart.App.client.Api;
import com.example.focusstartsecondpart.App.client.Client;
import com.example.focusstartsecondpart.features.events.domain.model.Event;

import java.util.List;

import io.reactivex.Single;
import io.reactivex.schedulers.Schedulers;

public class EventsLoaderImpl implements EventsLoader{

    private Api api;

    public EventsLoaderImpl(Api api){
        this.api = api;
    }

    @Override
    public Single<List<Event>> loadEvents() {
        return api.getEventList(Client.getInstance().getCftToken())
               .subscribeOn(Schedulers.io());
    }
}

File: app/src/main/java/com/example/focusstartsecondpart/features/events/data/network/EventsLoaderImpl.kt
package com.example.focusstartsecondpart.features.events.data.network

import com.example.focusstartsecondpart.App.client.Api
import com.example.focusstartsecondpart.App.client.Client
import com.example.focusstartsecondpart.features.events.domain.model.Event

import io.reactivex.Single
import io.reactivex.schedulers.Schedulers

class EventsLoaderImpl(private val api: Api) : EventsLoader {
    override fun loadEvents(): Single<List<Event>> {
        return api.getEventList(Client.getInstance().cftToken)
                .subscribeOn(Schedulers.io())
    }
}
